proto/connector/v1/connector.proto

```diff
@@ -54,6 +54,27 @@ service SourcePlugin {
   // other function. After Teardown returns, the plugin should be ready for a
   // graceful shutdown.
   rpc Teardown(Source.Teardown.Request) returns (Source.Teardown.Response);
+
+  // -- Lifecycle events -------------------------------------------------------
+
+  // LifecycleOnCreated is called after Configure and before Start when the
+  // connector is run for the first time. This call will be skipped if a
+  // connector was already started before. This method can be used to do some
+  // initialization that needs to happen only once in the lifetime of a
+  // connector (e.g. create a replication slot). Anything that the connector
+  // creates in this method is considered to be owned by this connector and
+  // should be cleaned up in LifecycleOnDeleted.
+  rpc LifecycleOnCreated(Source.Lifecycle.OnCreated.Request) returns (Source.Lifecycle.OnCreated.Response);
+  // LifecycleOnUpdated is called after Configure and before Start when the
```

Comment:
Can `Configure` be called on a running plugin?

Reply:
Do you mean that `Configure` would be called _after_ `Start` and `Run`? That won't happen. The other methods that are defined above will always be called in the order `Configure` -> `Start` -> `Run` -> `Stop` -> `Teardown` (or simply `Configure` -> `Teardown` in case we are validating the config).

Reply:
> Do you mean that `Configure` would be called _after_ `Start` and `Run`?

Yup, that's what I meant! Thanks for the clarification! :+1: